osc-server/src/test/java/org/osc/core/broker/service/servicefunctionchain/ServiceFunctionChainRequestValidatorTest.java

```diff
@@ -0,0 +1,276 @@
+
+		AddOrUpdateServiceFunctionChainRequest request = new AddOrUpdateServiceFunctionChainRequest();
+		BaseDto dto = new BaseDto();
+		String name = StringUtils.repeat("s", 15);
+		request.setName(name);
+		dto.setParentId(1L);
+		request.setDto(dto);
+		// Arrange.
+		this.exception.expect(VmidcBrokerValidationException.class);
+
+		// Act.
+		this.validator.validate(request);
+	}
+
+	@Test
+	public void testValidate_WithBeginingInvalidNameChar_ThrowsVmidcBrokerValidationException() throws Exception {
+
+		AddOrUpdateServiceFunctionChainRequest request = new AddOrUpdateServiceFunctionChainRequest();
+		BaseDto dto = new BaseDto();
+		request.setName("1sfc");
+		dto.setParentId(1L);
+		request.setDto(dto);
+		// Arrange.
+		this.exception.expect(VmidcBrokerValidationException.class);
+
+		// Act.
+		this.validator.validate(request);
+	}
+
+	@Test
+	public void testValidate_WithInvalidNameChar_ThrowsVmidcBrokerValidationException() throws Exception {
+
+		AddOrUpdateServiceFunctionChainRequest request = new AddOrUpdateServiceFunctionChainRequest();
+		BaseDto dto = new BaseDto();
+		request.setName("sfc**2");
+		dto.setParentId(1L);
+		request.setDto(dto);
+		// Arrange.
+		this.exception.expect(VmidcBrokerValidationException.class);
+
+		// Act.
+		this.validator.validate(request);
+	}
+
+	@Test
+	public void testValidate_WhenParentIdIsNull_ThrowsVmidcBrokerValidationException() throws Exception {
+		AddOrUpdateServiceFunctionChainRequest request = new AddOrUpdateServiceFunctionChainRequest();
+		BaseDto dto = new BaseDto();
+		request.setName("sfc-1");
+		request.setDto(dto);
+		// Arrange.
+		this.exception.expect(VmidcBrokerInvalidEntryException.class);
+
+		// Act.
+		this.validator.validate(request);
+	}
+
+	@Test
+	public void testValidate_WhenVcIdIsNotMatched_ThrowsVmidcBrokerValidationException() throws Exception {
+
+		registerVirtualConnector("NONE", 1L);
+		AddOrUpdateServiceFunctionChainRequest request = new AddOrUpdateServiceFunctionChainRequest();
+		BaseDto dto = new BaseDto();
+		request.setName("sfc-1");
+		dto.setParentId(2L);
+		request.setDto(dto);
+		// Arrange.
+		this.exception.expect(VmidcBrokerValidationException.class);
+
+		// Act.
+		this.validator.validate(request);
+	}
+
+	@Test
+	public void testValidate_WhenControllerTypeNotNone_ThrowsVmidcBrokerValidationException() throws Exception {
+
+		registerVirtualConnector("OTHER", 1L);
+		AddOrUpdateServiceFunctionChainRequest request = new AddOrUpdateServiceFunctionChainRequest();
+		BaseDto dto = new BaseDto();
+		request.setName("sfc-1");
+		dto.setParentId(1L);
+		request.setDto(dto);
+		// Arrange.
+		this.exception.expect(VmidcBrokerValidationException.class);
+
+		// Act.
+		this.validator.validate(request);
+	}
+
+	@Test
+	public void testValidate_WhenVsIdsNotExist_ThrowsVmidcBrokerValidationException() throws Exception {
+
+		registerVirtualConnector("NONE", 1L);
```

Comment:
i think this would cause the VmidcBrokerValidationException not the vs id not existing right? Please check the error message you are getting.

maybe we should validate the error message so we get the right error? @emanoelxavier thoughts?

Reply:
We are receiving VmidcBrokerValidationException  in this case when Vsids are doesn not exist.